assets/training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -263,14 +270,27 @@ def download_model(self):
                 "misc": model_props.get("misc"),
             }
         else:
-            raise AzureMLException._with_error(
+            raise AzureMLException._with_error(run_command
                 AzureMLError.create(InvalidHuggingfaceModelIDError, model_id=self._model_id)
             )
 
 
-def download_model(model_source: str, model_id: str, download_dir: Path):
+def download_model(model_source: str, model_id: str, download_dir: Path, auth_token: str):
     """Download model and return model information."""
     if model_source == ModelSource.HUGGING_FACE.value:
+        if auth_token:
+            try:
+                cmd = "git config --global credential.helper store"
+                exit_code, stdout = run_command(cmd)
+                if exit_code != 0:
+                    raise AzureMLException._with_error(
+                        AzureMLError.create(GITConfigError, error=stdout)
+                    )
+                login(token=auth_token, add_to_git_credential = True)
+            except Exception as ex:
+                raise AzureMLException._with_error(
+                    AzureMLError.create(HFAuthenticationError, error=ex)
+                )      
         downloader = HuggingfaceDownloader(model_id=model_id, download_dir=download_dir)
```

Comment:
lets perform logout after download operation is completed

Reply:
Done.